include/hip/hcc_detail/hip_fp16.h

```diff
@@ -29,6 +29,8 @@ THE SOFTWARE.
     #include <utility>
 #endif
 
+#include "hip/hcc_detail/hip_fp16.h"
```

Comment:
This looks weird. The header is explicitly including itself. Maybe the wrong include here?